src/Microsoft.AspNetCore.Diagnostics.HealthChecks/HealthCheckMiddleware.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.AspNetCore.Diagnostics.HealthChecks
+{
+    public class HealthCheckMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly HealthCheckOptions _healthCheckOptions;
+        private readonly IHealthCheckService _healthCheckService;
+
+        public HealthCheckMiddleware(RequestDelegate next, HealthCheckOptions healthCheckOptions, IHealthCheckService healthCheckService)
```

Comment:
IOptions

Reply:
Do our middlewares use IOptions? It looks like some in aspnet/BasicMiddleware do, but others such as Proxy and Authentication don't. Middleware don't come from DI, but I suppose they are activated from it. I can use IOptions.

Reply:
Yes, IOptions is pervasive now.